### DISCUSSION TITLE
HttpReader progress question

### DISCUSSION THREAD

**Discussion by AcidRaZor** [gildas-lormeau/zip.js#529] [Q&A]
Just stumbled upon this library and it seems to fit my use-case. I have files stored on a server, accessible through my API to logged in users. Content Disposition for these links to download 1 file is set to attachment (default inline)

Using HTTPReader, I was hoping to have the user download multiple selected files and ZIP them up to download just the ZIP as many users struggle with downloading single files (they find it tedious for obvious reasons, whereas a "select all then work with 1 ZIP" is much easier, even though technically they download each file to their browser in the background.

I currently use axios with a download progress callback to show them the % of download, does HttpReader have something similar to indicate the process of download before it's added to the ZIP?

**Comment by gildas-lormeau** [comment]
Did you try to use the [`onprogress`](https://gildas-lormeau.github.io/zip.js/api/interfaces/ZipWriterAddDataOptions.html#onprogress) option when calling [`ZipWriter#add()`](https://gildas-lormeau.github.io/zip.js/api/classes/ZipWriter.html#add)? It should fulfill your needs hopefully.

**Comment by AcidRaZor** [comment]
> Did you try to use the [`onprogress`](https://gildas-lormeau.github.io/zip.js/api/interfaces/ZipWriterAddDataOptions.html#onprogress) option when calling [`ZipWriter#add()`](https://gildas-lormeau.github.io/zip.js/api/classes/ZipWriter.html#add)? It should fulfill your needs hopefully.

Thanks, I'll check it out, hopefully it does!

**Comment by gildas-lormeau** [comment]
You will probably need to use Range requests as well (see https://gildas-lormeau.github.io/zip.js/api/interfaces/HttpOptions.html#useRangeHeader) otherwise the `onprogress` will only reflect the compression progress.

**Comment by AcidRaZor** [comment]
Perfect thanks, the backend team confirmed we won't be able to support range headers right now, but at least there's some progress indicator for the UX. Thanks so much!

**Reply by gildas-lormeau** [reply]
You're welcome. Note that if RAM is not a problem, you could also download the file as a `Blob` (for example) and use `onDownloadProgress` in axios to track the download. Then you could use a `BlobReader` with the `onprogress` callback to track the compression.
